File: src/commands.ts
import { bold, underline, yellow, blue, green, cyan } from "@std/fmt/colors";
import { Config } from "./config.ts";

export function showHelp() {
  console.log(bold(underline("Commander - The CLI to manage code challenges")));
  console.log(`\n  ${underline(bold(cyan("Usage:")))}`);
  console.log(`    ${blue("commander")} ${yellow("[options]")} ${green("[directory]")}`);
  console.log(`\n  ${underline(bold(cyan("Options:")))}`);
  console.log(`    ${yellow("-h")}, ${yellow("--help")}              Show this help message`);
  console.log(`    ${yellow("-i")}, ${yellow("--init")}              Create a default config file`);
  console.log(`    ${yellow("-c")}, ${yellow("--config=<path>")}     Specify config file path (default: config.toml)`);
  console.log(`    ${yellow("-d")}, ${yellow("--dir=<path>")}        Specify directory to process`);
  console.log(`    ${yellow("-l")}, ${yellow("--list")}              List supported languages and their configurations`);
  console.log(`\n  ${underline(bold(cyan("Examples:")))}`);
  console.log(`    ${blue("commander")} ${green("tests")}`);
  console.log(`    ${blue("commander")} ${yellow("--init")}`);
  console.log(`    ${blue("commander")} ${yellow("--dir=tests")} ${yellow("--config=custom.toml")}`);
}

export function listLanguages(config: Config) {
  console.log(bold(underline("Configured Languages:")));

  for (const lang of config.supportedExtensions) {
    const langConfig = config.getLanguageConfig(lang);
    if (langConfig) {
      console.log(bold(`\n.${lang} files:`));
      console.log(`  Runner: ${blue(langConfig.runner)}`);
      console.log(`  Flags: ${langConfig.flags.length ? blue(langConfig.flags.join(" ")) : "(none)"}`);
      console.log(`  Directories: ${langConfig.directories.length ? blue(langConfig.directories.join(", ")) : "(none)"}`);
    }
  }

  if (config.supportedExtensions.length === 2) {
    console.log(yellow("\nYou can add more languages by editing the config file. If you haven't created one yet, run `commander --init` to create one."));
  }
}

export async function createConfigFile(configPath: string): Promise<boolean> {
  const config = new Config(configPath);
  if (await config.save()) {
    console.log(yellow(`Created config file at ${configPath}`));
    return true;
  }
  return false;
}

File: internal/cmd/commands.go
package cmd

import (
	"fmt"
	"strings"

	"github.com/4ster-light/commander/internal/config"
	"github.com/fatih/color"
)

func ShowHelp() {
	bold := color.New(color.Bold).SprintFunc()
	green := color.New(color.FgGreen).SprintFunc()
	yellow := color.New(color.FgYellow).SprintFunc()
	blue := color.New(color.FgBlue).SprintFunc()

	fmt.Printf("%s\n\n", bold("Commander - The CLI to manage code challenges"))
	fmt.Println(bold("Usage:"))
	fmt.Printf("  %s %s %s\n\n",
		blue("commander"),
		yellow("[options]"),
		green("[directory]"))

	fmt.Println(bold("Options:"))
	fmt.Printf("  %-20s %s\n", yellow("-h"), "Show this help message")
	fmt.Printf("  %-20s %s\n", yellow("-i"), "Create default config file")
	fmt.Printf("  %-20s %s\n", yellow("-c=<path>"), "Specify config file path")
	fmt.Printf("  %-20s %s\n", yellow("-d=<path>"), "Specify directory to process")
	fmt.Printf("  %-20s %s\n", yellow("-l"), "List supported languages")

	fmt.Println(bold("\nExamples:"))
	fmt.Printf("  %s %s\n", blue("commander"), color.GreenString("scripts"))
	fmt.Printf("  %s %s\n", blue("commander"), yellow("-i"))
	fmt.Printf("  %s %s\n", blue("commander"), yellow("-d=scripts"))
}

func ListLanguages(cfg *config.Config) {
	bold := color.New(color.Bold).SprintFunc()

	fmt.Println(bold("Configured Languages:"))
	for lang, lc := range cfg.Data.Languages {
		fmt.Printf("\n%s files:\n", bold("."+lang))
		fmt.Printf("  Runner: %s\n", lc.Runner)
		fmt.Printf("  Flags: %s\n", formatFlags(lc.Flags))
		fmt.Printf("  Directories: %s\n", formatDirs(lc.Directories))
	}
}

func CreateConfig(path string) error {
	cfg := config.NewDefaultConfig()
	cfg.ConfigPath = path
	return cfg.Save()
}

func formatFlags(flags []string) string {
	if len(flags) == 0 {
		return "(none)"
	}
	return strings.Join(flags, " ")
}

func formatDirs(dirs []string) string {
	if len(dirs) == 0 {
		return "(none)"
	}
	return strings.Join(dirs, ", ")
}
